sqlmesh/dbt/loader.py

```diff
@@ -138,6 +139,8 @@ def _load_project(self) -> Project:
             ),
             variables=self._variables,
         )
+        if self._project.context.target.database != self._context.default_catalog:
+            raise ConfigError("Project default catalog does not match context default catalog")
```

Comment:
When can this happen?

Reply:
It should never happen. This is just making sure that isn't the case.